### PR TITLE
Profit

### PR DESCRIPTION
**Pull request by spalen0**

### CODE REVIEW

**Comment by spalen0** [comment]
image

File: Makefile [hunk 1]
```diff
@@ -9,16 +9,17 @@ size  :; forge build --sizes
 inspect :; forge inspect ${contract} storage-layout --pretty
 
 FORK_URL := ${ETH_RPC_URL} 
+FORK_BLOCK_NUMBER := 19435506
```
Comment (newmickymousse) [inline Makefile]: why are we fixing a block number?

File: src/YEthStakerStrategy.sol [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+    constructor(
+        address _asset,
+        string memory _name,
+        address _gov
+    ) BaseStrategy(_asset, _name) {
+        require(_asset == address(WETH), "Asset!=WETH");
+        WETH.approve(address(curvepool), type(uint256).max);
+        yETH.approve(address(curvepool), type(uint256).max);
+        yETH.approve(address(styETH), type(uint256).max);
+        require(_gov != address(0), "GOV=0x0");
```
Comment (newmickymousse) [inline src/YEthStakerStrategy.sol:67]: instead of gov let's just use the roles that are in v3: role_manager

Comment (newmickymousse) [reply src/YEthStakerStrategy.sol:67]: Please check: https://github.com/yearn/yearn-vaults-v3/blob/master/contracts/interfaces/Roles.sol

EmergencyManager

File: src/YEthStakerStrategy.sol [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+        );
+    }
+
+    /**
+     * @notice Internal function to calculate the value of different assets
+     * in asset value.
+     *
+     * @return estimated total value in asset value
+     */
+    function estimateTotalAssets() public view returns (uint256) {
```
Comment (newmickymousse) [inline src/YEthStakerStrategy.sol]: can we rename function to `estimatedTotalAssest()`?

**Review by newmickymousse — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (spalen0) [reply Makefile]: It's faster to run the tests when you are on a fixed number. Lower bandwidth. I'll remove it.

File: src/YEthStakerStrategy.sol (hunk 3 above)
Comment (spalen0) [reply src/YEthStakerStrategy.sol]: Renamed: https://github.com/newmickymousse/yeth-strategy/pull/2/files#diff-84b201488eaff9a264559f553c6629871e7e947dbd698d066d403365345d5c33R276

File: Makefile (hunk 1 above)
Comment (spalen0) [reply Makefile]: https://github.com/newmickymousse/yeth-strategy/pull/2/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R14

**Pull request merged**